### ISSUE TITLE
Possibility to export

### ISSUE THREAD

**Issue by throrin19** [andulv/RSTabExplorer#13]
It's possible to export viewed tabs into powetab or guitar pro or text representation ?

**Comment by DeathEvil** [comment]
+1, seems like the most important feature along with print

**Comment by fholger** [comment]
If you're looking for export to Guitar Pro, you might check out my alternative program: https://github.com/fholger/RocksmithToTab. I wrote it after I discovered the tab explorer and also thought that export features would be very useful :)

**Comment by DeathEvil** [comment]
Thanks for the heads up!

**Comment by DeathEvil** [comment]
Great tool, only wish there was gui

**Comment by fholger** [comment]
I know, sorry, I don't like gui programming :D 
For the most part, though, you can just drag'n'drop a psarc file (e.g. the main songs.psarc in the Rocksmith folder) onto the RocksmithToTab.exe in the explorer, and it will give you tabs for all the songs contained inside. So you don't really have to deal with the command line, unless you want to do something more specific.

**Comment by DeathEvil** [comment]
Yes, it's definitely not necessary. Thank you again so much for making the tool! So far only one file errors out and it's from new Duran Duran pack

**Comment by fholger** [comment]
I don't have that, unfortunately, but yeah, a few might not work at the moment. I've also received a few bug reports from other people, I'll try to fix as many as I can.
If you have Guitar Pro 6, you might try gpx output (in this case, you do have to go to the console and add -f gpx as a parameter). The gpx output is sometimes a little more forgiving.

**Comment by DeathEvil** [comment]
Is there an undocumented command line to convert all parts from a folder with GPS ext? Thanks

**Comment by DeathEvil** [comment]
Stupid autocorrect, I meant all psarcs from a folder with GPS ext.

**Comment by fholger** [comment]
No, sorry, the program only handles one psarc at the time. You could use a .bat file though. For example, create file ExportFolder.bat with the following contents:

```
@echo off
for /f "delims=" %%f in ('dir /b /a-d-h-s %1\*.psarc') do (
  c:\temp\RocksmithToTab\RocksmithToTab.exe -f gpx -o gpx_tabs "%1\%%f"
)
pause
```

You need to adjust the path the RocksmithToTab.exe to the full path on your PC. You can set any parameters you want, e.g. -f gpx to convert to gpx, and -o somefolder to specify an output folder. Then you can drop the folder onto that bat and it should convert all psarc inside. I hope...

**Comment by DeathEvil** [comment]
fholger, you are a legend! works like a charm:)

**Comment by DeathEvil** [comment]
ok, from RS1 on disc, RS1 DLC, RS2014 on disc, and RS2014 DLC, only 2 files from RS2014 DLC error out: "duranhun", and "swinglifeagain". I'd be happy to send you the psarcs for analysis, just pm with email.
thanks again for an outstanding soft:)

**Comment by fholger** [comment]
You're welcome. I have a few other bugs to figure out first, but do feel free and open a bug report for them over at the project :)

**Comment by throrin19** [comment]
with your script, i have error : 

```
Opening archive .\songs.psarc ...
Converting song alliwannado...
Opening arrangement bass for song id alliwannado...
sngFile is null.
  Failed to get arrangement bass
Opening arrangement combo for song id alliwannado...
sngFile is null.
  Failed to get arrangement combo
Opening arrangement combo2 for song id alliwannado...
sngFile is null.
  Failed to get arrangement combo2
Converting song batcountry...
Opening arrangement bass for song id batcountry...
sngFile is null.
  Failed to get arrangement bass
Opening arrangement lead1 for song id batcountry...
sngFile is null.
  Failed to get arrangement lead1
Opening arrangement lead2 for song id batcountry...
sngFile is null.
  Failed to get arrangement lead2
Opening arrangement rhythm for song id batcountry...
sngFile is null.
  Failed to get arrangement rhythm
Converting song blitzkriegbop...
Opening arrangement bass for song id blitzkriegbop...
sngFile is null.
  Failed to get arrangement bass
Opening arrangement combo for song id blitzkriegbop...
sngFile is null.
  Failed to get arrangement combo
Opening arrangement combo2 for song id blitzkriegbop...
sngFile is null.
  Failed to get arrangement combo2
Converting song bloodandthunder...
```

**Comment by fholger** [comment]
Are you using the Mac version? I think the Mac version packs the psarcs differently (that's why there's usually two versions for the DLCs), and you'd need a different version of the Rocksmith Toolkit for those, I think. I haven't tried it on Mac, so I can't currently help with that.

If it's not the Mac version, then it's very strange. Do you have any DLCs you could try? Try those ending in _p.psarc.

**Comment by throrin19** [comment]
yes, i'm on mac. Ok i don't know that. I try this night after download windows version. Thanks
